packages/create-sitecore-jss/src/templates/node-headless-ssr-proxy/package.json

```diff
@@ -32,5 +31,14 @@
     "memory-cache": "^0.2.0",
     "dotenv": "^16.0.0"
   },
+  "devDependencies": {
+    "@types/compression": "^1.7.2",
+    "@types/express": "^4.16.4",
+    "@types/memory-cache": "^0.2.2",
+    "@types/node": "^17.0.31",
+    "del-cli": "^4.0.1",
+    "ts-node": "^10.7.0",
+    "typescript": "^4.6.4"
```

Comment:
I suggest you to use `~`, because sometimes `typescript` can release breaking changes even within the new minor version